README.md

```diff
@@ -1,2 +1,352 @@
+# Deep Diving 
+
+Before starting, I want to clear some of my notations. I will abide to the conventions of Linux man pages. In those:
+
+`[ ]` means that anything within (any values, information or parameters) are optional. You can chose one, more than one or none at all.
+`< >` means that the element within (again any parameter, information or a value) is mandatory. We require to replace the text in between angle brackets with actual appropriate information.
+`-`means the options. It does not stand by itself, it will be followed by some form of a value ( or information or parameter).
+
+`-` We will not use that one I guess but while on it a little extra does not kill anyone ... It means long options like a more descriptive name of the option name above. It is pretty common and you may stumble upon them anywhere. For example an option `-l` indicating an option as a 'list', will be given `--list` as a long option as well. Both accomplish the same thing. When you use heavily on options, it helps to use long options for visibility and readability for you and others.
```

Comment:
> For example 

For example, *(comma is a must), also see line 11 for the same issue. Probably there are more, you can ctrl+f and search the whole document 🗡️*

Reply:
> We will not use that one I guess but while on it a little extra does not kill anyone

I suggest re-writing this sentence. Too much conjunctions:
- I guess
- but
- while

also, *while on it* is a bit unclear. Maybe *while we're on it*, but I still suggest rewriting this from scratch.

Reply:
> It means long options

*It is the shorthand version of the long options* 
Is it what you mean? If so, `it means long options` actually confused me :D

Reply:
> > We will not use that one I guess but while on it a little extra does not kill anyone
> 
> I suggest re-writing this sentence. Too much conjunctions:
> 
> * I guess
> * but
> * while
> 
> also, _while on it_ is a bit unclear. Maybe _while we're on it_, but I still suggest rewriting this from scratch.

Yes you are absolutely right. I am rewriting and dividing it to parts.

Reply:
Yes I understood it correctly then. You confused me on something I already know haha